sky/serve/core.py

```diff
@@ -466,6 +468,36 @@ def down(
     sky_logging.print(stdout)
 
 
+@usage_lib.entrypoint
+def terminate_replica(service_name: str, replica_id: int) -> None:
+    cluster_status, handle = backend_utils.is_controller_up(
+        controller_type=controller_utils.Controllers.SKY_SERVE_CONTROLLER,
+        stopped_message='All services should have terminated.')
+    if handle is None or handle.head_ip is None:
+        # The error message is already printed in
+        # backend_utils.is_controller_up
+        # TODO(zhwu): Move the error message into the exception.
+        with ux_utils.print_exception_no_traceback():
+            raise exceptions.ClusterNotUpError(message='',
+                                               cluster_status=cluster_status)
+
+    backend = backend_utils.get_backend_from_handle(handle)
+    assert isinstance(backend, backends.CloudVmRayBackend)
+
+    code = serve_utils.ServeCodeGen.terminate_replica(service_name, replica_id)
+    returncode = backend.run_on_head(handle,
+                                     code,
+                                     require_outputs=False,
+                                     stream_logs=False)
+    subprocess_utils.handle_returncode(
```

Comment:
Pls try-catch and re-raise RuntimeError here; ref https://github.com/skypilot-org/skypilot/blob/073e1898066844252b3bf8d0bf1e089b231a8ebc/sky/serve/core.py#L378-L390